platform/addons.cue

```diff
@@ -1,43 +1,39 @@
 package holos
-import ("encoding/json")
 
+import ("encoding/json")
 
-Addons: {
-  "cilium": {
-    path: "addons/cni/cilium"
-    schema: {
-      fleet: {
-        name: string
-      }
-      addonParam: string
-    }
-    parameters: {
-      addonParam: "test"
-    }
-    selector: {
-      "cluster_cni": "cilium"
-    }
-  }
+// hidden because parameters is not concrete.
+_Addons: {
+	"cilium": {
+		path:       "addons/cni/cilium"
+		parameters: #CiliumParameters
+		selector: {
+			"cluster_cni": "cilium"
+		}
+	}
 }
 
 // === Render ===
 
 for fleetName, fleet in Fleets {
-  for clusterName, cluster in fleet.clusters{
-    for addonName, addon in Addons {
-      _params: addon.schema & {
-        "fleet": fleet.parameters
-        "cluster": cluster.parameters
-      } & addon.parameters
-
-      Platform: Components: {
-        "\(fleetName).\(clusterName).addons.\(addonName)": {
-          name: "\(fleetName).\(clusterName).addons.\(addonName)"
-          path: addon.path
-          writeTo?: fleet.componentWriteTo
-          parameters: holos_params: json.Marshal(_params)
-        }
-      }
-    }
-  }
+	for clusterName, cluster in fleet.clusters {
+		for addonName, addon in _Addons {
+			Platform: Components: {
+				"\(fleetName).\(clusterName).addons.\(addonName)": {
+					let FLEET = fleet
+					let CLUSTER = cluster
```

Comment:
let avoids name collisions when referring to fleet.parameters and cluster.paramers